pythonFiles/vscode_pytest/__init__.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+import enum
+import json
+import os
+import pathlib
+import sys
+from typing import List, Literal, Tuple, TypedDict, Union
+
+import pytest
+from _pytest.doctest import DoctestItem, DoctestTextfile
+
+script_dir = pathlib.Path(__file__).parent.parent
+sys.path.append(os.fspath(script_dir))
+sys.path.append(os.fspath(script_dir / "lib" / "python"))
+
+
+# Inherit from str so it's JSON serializable.
+class TestNodeTypeEnum(str, enum.Enum):
+    class_ = "class"
+    file = "file"
+    folder = "folder"
+    test = "test"
+    doc_file = "doc_file"
+
+
+class TestData(TypedDict):
+    name: str
+    path: str
+    type_: TestNodeTypeEnum
+    id_: str
+
+
+class TestItem(TestData):
+    lineno: str
+    runID: str
+
+
+class TestNode(TestData):
+    children: "List[Union[TestNode, TestItem]]"
+
+
+from testing_tools import socket_manager
+from typing_extensions import NotRequired
+
+DEFAULT_PORT = "45454"
+
+
+def pytest_collection_finish(session):
+    # Called after collection has been performed.
+    node: Union[TestNode, None] = build_test_tree(session)[0]
+    cwd = pathlib.Path.cwd()
+    if node:
+        sendPost(str(cwd), node)
+    # TODO: add error checking.
+
+
+def build_test_tree(session) -> Tuple[Union[TestNode, None], List[str]]:
+    # Builds a tree of tests from the pytest session.
+    errors: List[str] = []
+    session_node: TestNode = create_session_node(session)
+    session_children_dict: dict[str, TestNode] = {}
```

Comment:
Need to use `Dict` like you used `List` above.

```suggestion
    session_children_dict: Dict[str, TestNode] = {}
```